impl.go

```diff
@@ -111,10 +166,197 @@ func findInterface(iface string, srcDir string) (path string, id string, err err
 	}
 	decl := f.Decls[1].(*ast.GenDecl)      // var i io.Reader
 	spec := decl.Specs[0].(*ast.ValueSpec) // i io.Reader
-	sel := spec.Type.(*ast.SelectorExpr)   // io.Reader
-	id = sel.Sel.Name                      // Reader
+	iface, err = typeFromAST(spec.Type)
+	if err != nil {
+		return path, iface, fmt.Errorf("error parsing type from AST: %w", err)
+	}
+	// trim off the package which got smooshed on when resolving the type
+	_, iface.ID, _ = strings.Cut(iface.ID, ".")
+	return path, iface, err
+}
+
+func typeFromAST(in ast.Expr) (Type, error) {
+	switch specType := in.(type) {
```

Comment:
is there any way to avoid rebuilding an entire node printing infrastructure? it'd be nice to somehow use go/format or go/printer.